test/url-to-options.js

```diff
@@ -0,0 +1,56 @@
+import url from 'url';
+import test from 'ava';
+import urlToOptions from '../source/url-to-options';
+
+const githubURL = 'https://user:password@github.com:443/';
+
+test('converts node legacy URL to options', t => {
+	const parsedURL = url.parse(githubURL);
+	const options = urlToOptions(parsedURL);
+	const expected = {
+		hash: null,
+		hostname: 'github.com',
+		href: githubURL,
+		path: '/null',
```

Comment:
This doesn't seem right 🤔 .

Reply:
Smart!

`url.search` is null in this case. It should add that to `path`. Instead:

https://github.com/alextes/got/blob/dccde51438af10a8a5236288c52b643dd95eb969/source/url-to-options.js#L11

do:
```js
	const options = {
		protocol: url.protocol,
		hostname: url.hostname.startsWith('[') ? url.hostname.slice(1, -1) : url.hostname,
		hash: url.hash,
		search: url.search,
		pathname: url.pathname,
		path: `${url.pathname}`,
		href: url.href
	};
	if (url.search !== null) {
		options.path += url.search;
	}
```